### PR TITLE
Router clean up

### PR DESCRIPTION
**Pull request by alwx**
Works together with https://github.com/status-im/status-go/pull/5778

### Summary

Updates router calls to work with changes made in https://github.com/status-im/status-go/pull/5778

### CODE REVIEW

File: src/status_im/contexts/wallet/send/events.cljs
Comment (briansztamfater) [inline src/status_im/contexts/wallet/send/events.cljs]: On the previous status-go version, there was a typo in the endpoint name (Calcualtion). Now is fixed on status-go side, so this should be `wallet_stopSuggestedRoutesAsyncCalculation`

File: src/status_im/contexts/wallet/swap/events.cljs
Comment (dlipicar) [inline src/status_im/contexts/wallet/swap/events.cljs]: typo here as well

**Comment by Khushboo-dev-cpp** [comment]
@alwx I am working on a PR that has api change https://github.com/status-im/status-go/pull/5564/files

basically removed 
```
        TokenPrice            *float64              `json:"TokenPrice,omitempty"`
	NativeChainTokenPrice *float64              `json:"NativeChainTokenPrice,omitempty"`
```
 and added 
`` 	UpdatedPrices map[string]float64    `json:"UpdatedPrices,omitempty"`` `
instead. 

I wanted to check if the removed params are being used in the mobile client already? cause if not I'll merge my PR to Sales PR directly. 

fyi @saledjenic

**Comment by saledjenic** [comment]
@alwx for testing this branch could you make it to point to statusgo commit `900ad9be01de8bfce2c229c8fb0222b99606cf35` cause I rebased it today?

**Comment by saledjenic** [comment]
@alwx what about Khushboo's question? Do you use those properties or not? If not we can merge here changes in the same go.

**Comment by alwx** [comment]
@saledjenic will do in a sec
Regarding the @Khushboo-dev-cpp's question: no, those parameters are not used anywhere.

**Review by smohamedjavid — APPROVED** [review]

**Comment by saledjenic** [comment]
@churik please let us know when is the right time to merge statusgo PR?

**Comment by churik** [comment]
when we'll test it, I'll ping you @saledjenic

**Comment by VolodLytvynenko** [comment]
hi @alwx thank you for PR. Take a look at the found issue. This issue happen consistently, but unfortunately there are no exact steps to reproduce it. Hopefully, the logs will help

## ISSUE 1: [Android] 'Undefined not a function' error is constantly shown after attempt to build the route

### Steps:
1. Go to routes generation page
2. Build the route
3. Go to transaction confirmation page
4. Go back to the wallet main page

### Actual result:
'Undefined not a function' error is constantly shown

https://github.com/user-attachments/assets/2fd26a54-8f4e-405f-8c78-9ce4d3079fb9


### Expected result:
no errors are shown

### Devices:
- Pixel 7a, Android 13

### Logs
[Status-debug-logs.zip](https://github.com/user-attachments/files/16945706/Status-debug-logs.zip)

**Comment by VolodLytvynenko** [comment]
## PR_ISSUE 2: Endless 'edit' button is shown 

### Steps:
1. Navigate to the route generation flow.
2. Input invalid data, such as:
  - A token count available on multiple networks.
  - A  zero value (e.g., 0.00000).
  - A value that cannot be covered due to insufficient fees.
3. Tap the "Try Again" button.

### Actual Result:
The "Edit" button is displayed repeatedly

https://github.com/user-attachments/assets/fdd89eec-3b0c-4eb8-b410-3810233a97e8


### Expected result:
The "Edit" button should be displayed only once

### OS: 
IOS, Android
### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17
### Logs
[logs (5).zip](https://github.com/user-attachments/files/16946542/logs.5.zip)

**Comment by shivekkhurana** [comment]
The edit button issue has been logged here already: https://github.com/status-im/status-mobile/issues/21043

I don't think it should be a scope of this PR.

**Comment by shivekkhurana** [comment]
@alwx Can you check the logs and confirm or deny that this issue is because of the changes you made ?

**Comment by VolodLytvynenko** [comment]
> The edit button issue has been logged here already: #21043
> 
> I don't think it should be a scope of this PR.

@shivekkhurana Yes, the first [issue](https://github.com/status-im/status-mobile/issues/21043) and the [current one](https://github.com/status-im/status-mobile/pull/21235#issuecomment-2341058497) seem similar, but this one appears to be easier to reproduce. Additionally, the endless button keeps appearing even without tapping the 'try again' button, so it might have a different cause.

**Comment by VolodLytvynenko** [comment]
> ## PR_ISSUE 2: Endless 'edit' button is shown
> ### Steps:
> 1. Navigate to the route generation flow.
> 2. Input invalid data, such as:
> 
> * A token count available on multiple networks.
> * A  zero value (e.g., 0.00000).
> * A value that cannot be covered due to insufficient fees.
> 
> 3. Tap the "Try Again" button.
> 
> ### Actual Result:
> The "Edit" button is displayed repeatedly
> 
>  editbutton.mp4 
> ### Expected result:
> The "Edit" button should be displayed only once
> 
> ### OS:
> IOS, Android
> 
> ### Devices:
> * Pixel 7a, Android 13
> * iPhone 11 Pro Max, IOS 17
> 
> ### Logs
> [logs (5).zip](https://github.com/user-attachments/files/16946542/logs.5.zip)

Here are some additional steps to reproduce this issue, which does not happen in the Develop.
### Steps:
1. Navigate to the route generation flow.
2. Input invalid data, such as:
- A zero value (e.g., 0.00000).
3. Go back
4. Navigate to routes generation page again

https://github.com/user-attachments/assets/c1d2a4ba-4ad1-4ba9-ae4a-36f6fe931eed

**Comment by VolodLytvynenko** [comment]
## PR_ISSUE 3: Wrong UI is shown after changing the token on the routes generation page

### Steps:
1. Go to routes generation page
2. Build the route
3. Tap the asset to open asset list
4. Change the asset

### Actual result:
The previous route is still shown

https://github.com/user-attachments/assets/d2f942eb-58f4-4c6f-b41c-9b9ff2bdb9ca


### Expected result:
The previous route is not shown when asset is changed


### OS: 
IOS, Android
### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

**Comment by VolodLytvynenko** [comment]
### PR_ISSUE 4: The Confirmation Keyboard Auto-Closes After Entering the Password
### Steps:
1. Navigate to the transaction confirmation page.
2. Attempt to enter the password.

### Actual Result:
The confirmation keyboard constantly auto-closes, preventing the user from entering the password properly.

Pixel 7a, Android 13

https://github.com/user-attachments/assets/450a660b-9cec-4d32-92d7-02dccdb7c653

iPhone 11 Pro Max, IOS 17

https://github.com/user-attachments/assets/53b48b90-82fd-4948-a97b-cac95b5012fb

### Expected Result:
The keyboard should remain open, allowing the user to enter the password without interruptions

### Logs:
[logs (6).zip](https://github.com/user-attachments/files/16947601/logs.6.zip)

**Comment by VolodLytvynenko** [comment]
## PR_ISSUE 5: "eq() not a number: <0.01" Error shown after transaction confirmation

### Steps:
1. Open account
2. Go to the routes generation page -> confirm a transaction

### Actual result:
The user is redirected to the activity page, where the error message "eq() not a number: <0.01" is displayed.

image


### Expected result:
The transaction should be confirmed without showing any error messages
### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

### Logs
[logs (7).zip](https://github.com/user-attachments/files/16947924/logs.7.zip)

**Comment by saledjenic** [comment]
@VolodLytvynenko @alwx I'm wondering if all those issues come from the updated statusgo or they were in the mobile app even before statusgo changes?

**Comment by smohamedjavid** [comment]
@VolodLytvynenko

> PR_ISSUE 2: Endless 'edit' button is shown

Regarding additional steps posted in https://github.com/status-im/status-mobile/pull/21235#issuecomment-2341254191, The routes are not supposed to be fetched if the user enters 0 or 0.000000. It's fixed here: https://github.com/status-im/status-mobile/pull/21209 (regression from https://github.com/status-im/status-mobile/pull/21136)

 > PR_ISSUE 3: Wrong UI is shown after changing the token on the routes generation page

This issue is fixed here: https://github.com/status-im/status-mobile/pull/21201

**Comment by alwx** [comment]
@saledjenic I don't think this issues have to do something with the `status-go` update. Your changes done there are a refactoring, you haven't changed the functionality, and what's broken here is mostly UI issues which have nothing to do with either your changes or this particular PR.

It's hard for me to imagine how `status-go` changes might lead to some issues with edit button or keyboard hiding automatically.

**Comment by saledjenic** [comment]
@churik @VolodLytvynenko @alwx can we merge then the statusgo changes, cause we've been waiting for them 3 weeks already? 

Then you can merge this PR that handles endpoint names changes for the mobile app, and you can address all those above-mentioned issues (since they were here even before statusgo changes) through other PRs and don't block the desktop app by waiting for them to be fixed in this PR?

**Comment by VolodLytvynenko** [comment]
## PR_ISSUE 6: "Native token not found" error message randomly occurs

Unfortunately, this issue doesn't occur consistently, and there are no exact steps to reproduce it. However, it seems to happen after attempting to find routes upon entering valid data into the input field. Hopefully, the logs will help.

### Steps:
1. Enter valid data into routes generation page to build the routes

### Actual result:
"Native token not found" error message is shown
image

### Expected result:
No errors are shown, routes are build

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

### Logs
[Status-debug-logs (2).zip](https://github.com/user-attachments/files/16959963/Status-debug-logs.2.zip)

**Comment by VolodLytvynenko** [comment]
> @churik @VolodLytvynenko @alwx can we merge then the statusgo changes, cause we've been waiting for them 3 weeks already?

Hey @saledjenic, I need some additional time to finish testing. Just 2-3 more hours

**Comment by VolodLytvynenko** [comment]
### PR_ISSUE 7: "Not enough native balance, token ETH, chain id <chainId>" Toast Shown When There Is Insufficient ETH to Cover the Fee

### Steps:
1. Go to the transaction confirmation page.
2. Enter the valid amount of ERC-20 requiring ETH to cover the fee.
3. Ensure the ETH balance is insufficient to cover the transaction fee.


### Actual Result:
A toast message is displayed: "Not enough native balance, token ETH, chain id <chainId>."

image


### Expected Result:
The toast message is not shown

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

**Comment by VolodLytvynenko** [comment]
@saledjenic Thank you for your work and patience.  I've finished testing. If [issue 6](https://github.com/status-im/status-mobile/pull/21235#issuecomment-2342923420) isn't Status Go-related, then the status go PR can be merged. I agree with @alwx that the other issues don't seem to be related to status go

**Comment by churik** [comment]
@alwx 
not great that it was merged 
1) without e2e
2) without clear desicions what to do with issues 
3) without `Tested-OK`

I understand that it will be not working without new router calls but we should decide how to deal with issues before merging the PR.

**Comment by alwx** [comment]
@churik my mistake, sorry — I somehow read the comment from @VolodLytvynenko above as an invitation to merge this one ☹️ Would be the right way to proceed with that now? Revert?

**Pull request merged**
